### ISSUE TITLE
[New rule] Link can be simplified to autolink

### ISSUE THREAD

**Issue by Marcono1234** [DavidAnson/markdownlint#753]
### Problem
Sometimes, maybe due to usage of graphical editors, users write overly verbose [inline links](https://spec.commonmark.org/0.30/#inline-link) and [reference links](https://spec.commonmark.org/0.30/#reference-link) where the link display text is the same as the link destination. For example:
```md
[https://example.com/my-page](https://example.com/my-page)
```

```md
[https://example.com/my-page][site]

[site]: https://example.com/my-page
```

Especially for complex URLs such Markdown code can be difficult to read, and it is error-prone in case only the display text is updated but the link destination is left unchanged and vice versa.

### Suggested rule behavior
In these cases markdownlint should suggest usage of [autolinks](https://spec.commonmark.org/0.30/#autolink), e.g. `<https://example.com/my-page>`. Autolinks make it much easier to read the Markdown code.

Corner cases to consider:
- The rule should only apply when the link target is actually a valid autolink (e.g. it should not apply for relative links)
- Special escaping rules for autolinks have to be considered, or the rule should not apply when special escaping is needed to convert the link to an autolink (because that might defeat the point about being less error-prone)
- The rule cannot apply when the link has additional data such as a hover title

**Comment by nschonni** [comment]
Not entirely the same, but the conversion of text to auto links is already covered by md034 autofixing

**Comment by DavidAnson** [comment]
Agreed!

**Comment by DavidAnson** [comment]
Reopening this issue because #399 is fixed (in `next` branch), but does not include this behavior. I'm still thinking about whether it should or if this is something different.
